File: pset2/crack.c
#define _XOPEN_SOURCE
#define SALT "50"

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>


int main(int argc, string argv[])
{    
    if (argc != 2)    
    {        
        printf("rofl\n");        
        return 1;
    }
        char *hash = argv[1];
        char pass[5];     
        for ( char i= '0' ; i <= 'z';i++)
        {   
            pass[0] = i;
            pass[1] = '\0';
            if (strcmp(hash, crypt(pass, SALT)) == 0)
            {
                printf("Found password: %s\n", pass);
                return 0;
            }
                    for ( char j = '0' ; j <= 'z';j++)
                    {       
                        pass[1] = j;
                        pass[2] = '\0';
                        if (strcmp(hash, crypt(pass, SALT)) == 0)
                        {
                            printf("Found password: %s\n", pass);
                            return 0;
                    }
                        for ( char k = '0' ; k <= 'z'; k++ )
                        {   
                            pass[2] = k;
                            pass[3] = '\0';
                            if (strcmp(hash, crypt(pass, SALT)) == 0)
                            {
                                printf("Found password: %s\n", pass);
                                return 0;
                            }
                                for ( char l = '0'; l <= 'z'; l++)
                                {         
                                    pass[3] = l;
                                    pass[4] = '\0';
                                    if (strcmp(hash, crypt(pass, SALT)) == 0)
                                    {
                                        printf("Found password: %s\n", pass);
                                        return 0;
                                    }
                                }      
                            }             
        }
    }
    printf("Could not found password\n");
    return 2;
}

File: pset6/crack.py
import cs50
import sys
import crypt

salt = "50"
letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz'

if len(sys.argv) != 2:
    print(" Usage:python crack.py hash")
    exit(1) 
    
# if the password is one character
for c in letters: 
    key = c
    result = crypt.crypt(key, salt)
    if result == sys.argv[1]:
        print("{}".format(key))
        exit(0)
       
# if the password is two characters
for c in letters: 
    key21 = c
    for c in letters:
        key2 = key21 + c
        result = crypt.crypt(key2, salt)
        if result == sys.argv[1]:
            print("{}".format(key2))
            exit(0)
                
# if the password is three characters
for c in letters: 
    key31 = c
    for c in letters:
        key32 = c
        for c in letters:
            key3 = key31 + key32 + c
            result = crypt.crypt(key3, salt)
            if result == sys.argv[1]:
                print("{}".format(key3))
                exit(0)
                   
# if the password is four characters
for c in letters: 
    key41 = c
    for c in letters:
        key42 = c
        for c in letters:
            key43 = c
            for c in letters:
                key4 = key41 + key42 + key43 + c
                result = crypt.crypt(key4, salt)
                if result == sys.argv[1]:
                    print("{}".format(key4))
                    exit(0) 
